airflow/www/views.py

```diff
@@ -3251,8 +3251,7 @@ def duration(self, dag_id: str, session: Session = NEW_SESSION):
                 y=scale_time_units(cumulative_y[task_id], cum_y_unit),
             )
 
-        dates = sorted({ti.execution_date for ti in task_instances})
-        max_date = max(ti.execution_date for ti in task_instances) if dates else None
+        max_date = max((ti.execution_date for ti in task_instances), default=None)
```

Comment:
Begs the question why we needed to sort the dates in the first place. Leftover code maybe, who knows.